Convert default recipe to custom resource with idempotence and make service parameters match exactly

axelrtgs opened the pull request:
fixes for #25 & #26

axelrtgs commented on resources/install.rb:
```diff
@@ -0,0 +1,34 @@
+    checksum new_resource.sha256
+    action :unzip
+    notifies :create, 'remote_file[install nssm]', :immediately
+    not_if { Digest::SHA256.file("#{Chef::Config[:file_cache_path]}/#{basename}.zip").hexdigest == new_resource.sha256 }
```
Hoping the tests pass since i didnt test it when the file wasnt present. Im sure theres a better way to achieve this. your input would be appreciated.

axelrtgs commented on resources/install.rb:
```diff
@@ -0,0 +1,34 @@
+    checksum new_resource.sha256
+    action :unzip
+    notifies :create, 'remote_file[install nssm]', :immediately
+    not_if { Digest::SHA256.file("#{Chef::Config[:file_cache_path]}/#{basename}.zip").hexdigest == new_resource.sha256 }  if ::File.exist?("#{Chef::Config[:file_cache_path]}/#{basename}.zip")
```
OK so this at least works on fresh install but i would like some feedback if this method makes sense

axelrtgs commented on resources/install.rb:
```diff
@@ -0,0 +1,34 @@
+    checksum new_resource.sha256
+    action :unzip
+    notifies :create, 'remote_file[install nssm]', :immediately
+    not_if { Digest::SHA256.file("#{Chef::Config[:file_cache_path]}/#{basename}.zip").hexdigest == new_resource.sha256 } if ::File.exist?("#{Chef::Config[:file_cache_path]}/#{basename}.zip")
```
OK so this at least works on fresh install but i would like some feedback if this method makes sense to ensure the resource doesn't run on every converge

dhoer replied:
I haven't looked at the windows_zipfile code, but I would have assumed that it would manage this for you.  So if file was already downloaded and checksum is the same, then don't execute.

axelrtgs commented:
sorry for the commit spam. i had to edit the files directly in git and not locally. feel free to squash the commits into one.

poblahblahblah commented:
Hey @axelrtgs - I recently created https://github.com/dhoer/chef-nssm/pull/29, which would allow an attribute to decide whether or not to install nssm. In our infrastructure we have nssm being installed by a chocolatey package, so this cookbook installing nssm would break a few things for us. 

Is this something you want to implement in this PR?

axelrtgs commented:
@poblahblahblah Implemented. was just a quick change and seemed to make sense. it doesn't change the default behavior either so its a great addition. thanks for the improvement

dhoer commented:
I guess that assumption is incorrect based on above discussion...

dhoer commented:
I will try to get this merged soon.

dhoer commented:
Thanks for the contribution!

axelrtgs commented:
My pleasure. Let me know if you have any comments

axelrtgs commented:
Just for tracking this merge should close #28 also

poblahblahblah commented:
@axelrtgs Did 3d7c34b1daa7861f19a67dcacfb27f8b2cbc80cf rollback the fix for #28?

axelrtgs commented:
@poblahblahblah looks like I'm finding some odd edge cases with the resetting parameters to default. it could be the result of some butchered testing doing some strnage things on my machines i run the cookbook on since its a live lab environment and that caused issues but could you please review the changes i made and see if you find any weird stuff. i had some troubles with a string vs symbol thing previously that i fixed.

Also i am not good at writing tests if you could help with that i would appreciate it. I added you to my repo to help if you can.

dhoer commented:
@axelrtgs Released as version to 4.0.0.  Oops, just caught the testing comment. Please add tests in another PR.  Thanks guys for making this cookbook better.
